pkg/http/server.go

```diff
@@ -381,3 +413,164 @@ func (s *Server) HTTPSURL() string {
 	}
 	return ""
 }
+
+func (s *Server) existCertificate(uDomain string) bool {
+	if _, err := os.Stat("./lets_encrypt/" + uDomain + ".key"); os.IsNotExist(err) {
+		return false
+	}
+	if _, err := os.Stat("./lets_encrypt/" + uDomain + ".crt"); os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+func (s *Server) getLetsEncrypt(uDomain string) error {
```

Comment:
this function is really long, let's break it down into smaller pieces. take a look at `crypto.go`, maybe we can share some logic with it.